mindsdb_native/libs/controllers/predictor.py

```diff
@@ -298,19 +289,24 @@ def predict(self,
         :return: TransactionOutputData object
         """
         with MDBLock('shared', 'learn_' + self.name):
-            if unstable_parameters_dict is None:
-                unstable_parameters_dict = {}
+            if advanced_args is None:
+                advanced_args = {}
 
             if run_confidence_variation_analysis is True and when_data is not None:
                 error_msg = 'run_confidence_variation_analysis=True is a valid option only when predicting a single data point via `when`'
                 self.log.error(error_msg)
                 raise ValueError(error_msg)
 
             transaction_type = TRANSACTION_PREDICT
-            when_ds = None if when_data is None else getDS(when_data)
-
-            # lets turn into lists: when
-            when = [when] if isinstance(when, dict) else when if when is not None else []
+            when_ds = None
+            when = None
```

Comment:
No it's not:

```
            elif self.transaction.hmd['model_when_conditions'] is not None:

                # if no data frame yet, make one
                df = self._get_data_frame_from_when_conditions()
```

Reply:
As in, `_get_data_frame_from_when_conditions` is only ever called *after* we figure out `model_when_conditions` is not `None`

Reply:
Buuut actually, good point, if `when` is always a list, why make it `None`, maybe I'm gonna break something. let me make it a list just in case :)

Reply:
Ok, I ended up keeping the default value of `when` as `None` but I simplified some of the data extractor logic so that it's more obvious that it doesn't cause issues.

I also realized I could now make `when_data` a mandatory argument => can avoid some check for the case where both `when` and `when_data` are empty.